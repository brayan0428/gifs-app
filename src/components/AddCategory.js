import React, { useState } from "react";
import PropTypes from "prop-types";
export const AddCategory = ({ addCategory }) => {
  const [inputValue, setInputValue] = useState("");

  const handleInputValue = (e) => {
    setInputValue(e.target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    if (inputValue.trim().length === 0) {
      return;
    }
    addCategory(inputValue);
    setInputValue("");
  };

  return (
    <form onSubmit={handleSubmit}>
      <input type="text" value={inputValue} onChange={handleInputValue} />
    </form>
  );
};

AddCategory.propTypes = {
  addCategory: PropTypes.func.isRequired,
};
