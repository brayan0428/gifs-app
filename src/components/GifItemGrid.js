import React from "react";
import PropTypes from "prop-types";

export const GifItemGrid = ({ img, title }) => (
  <div className="GifItemCard">
    <img className="GifItemCard__img" src={img} alt={title} />
    <div className="GifItemCard__body">
      <h4>{title}</h4>
    </div>
  </div>
);

GifItemGrid.propTypes = {
  img: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
};
