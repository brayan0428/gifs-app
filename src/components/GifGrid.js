import React from "react";
import { GifItemGrid } from "./GifItemGrid";
import useFetchGifs from "../hooks/useFetchGifs";
import PropTypes from "prop-types";

export const GifGrid = ({ category }) => {
  const { gifs, loading } = useFetchGifs(category);
  return (
    <>
      <h2>{category}</h2>
      {loading && <h4>Loading...</h4>}
      <div className="GifGrid">
        {gifs.map((gif) => (
          <GifItemGrid key={gif.id} {...gif} />
        ))}
      </div>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
