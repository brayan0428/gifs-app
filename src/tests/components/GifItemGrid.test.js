import React from "react";
import { shallow } from "enzyme";
import { GifItemGrid } from "../../components/GifItemGrid";

describe("<GifItemGrid />", () => {
  const title = "Prueba";
  const url = "https://prueba.com";
  const wrapper = shallow(<GifItemGrid title={title} img={url} />);
  test("Debe mostrar el componente correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe mostrar un h4 con el title de la prop", () => {
    const h4 = wrapper.find("h4");
    expect(h4.text().trim()).toBe(title);
  });

  test("La imagen debe tener el src y el alt", () => {
    const img = wrapper.find("img");
    expect(img.prop("src")).toBe(url);
    expect(img.prop("alt")).toBe(title);
  });
});
