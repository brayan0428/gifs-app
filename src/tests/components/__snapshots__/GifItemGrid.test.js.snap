// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifItemGrid /> Debe mostrar el componente correctamente 1`] = `
<div
  className="GifItemCard"
>
  <img
    alt="Prueba"
    className="GifItemCard__img"
    src="https://prueba.com"
  />
  <div
    className="GifItemCard__body"
  >
    <h4>
      Prueba
    </h4>
  </div>
</div>
`;
